clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -818,32 +822,29 @@ class SingleDeviceFunctionTracker {
                             DirectlyCalled);
     }
 
-    // Calculate the kernel body.  Note the 'isSYCLKernelBodyFunction' only
-    // tests that it is operator(), so hopefully this doesn't get us too many
-    // false-positives.
-    if (isSYCLKernelBodyFunction(CurrentDecl)) {
```

Comment:
I'm not quite following why the single check on `isSYCLKernelBodyFunction` was removed and now the same is checked in both branches of the `if` condition.